### ISSUE TITLE
Enqueue TinyMCE as a dependency

### ISSUE THREAD

**Issue by oscarssanchez** [QuizandSurveyMaster/quiz_master_next#1285]
**IF YOU DO NOT FOLLOW THIS TEMPLATE, YOUR ISSUE MAY BE CLOSED!!**

Please provide the following information when creating your issues:

## Site Info
WordPress Version: 5.7
QSM Version: 7.1.14
Browser: Chrome

## General description
If this is a bug, give a general description of what you are experiencing. If this is a feature request, include what the feature is that you are requesting.

Hello. 

We maintain the Brightcove Video Connect Plugin and noticed an issue when the plugin and Quiz Master are both active. Basically, the editor in Quiz Master is unusable.

## Expected behavior
If you are reporting a bug, please describe what you are expecting the plugin to do. If this is a feature request, delete this section.

You should be able to use the editor as normal.

## Actual behavior
If you are reporting a bug, please describe what you are experiencing. If this is a feature request, delete this section.

Editor is unusable.

## Steps to reproduce the behavior
If you are reporting a bug, describe the exact steps you are taking that results in the behavior described above. If this is a feature request, delete this section.

 Below are steps to reproduce:

1.- Install Quiz master
2.- Install Brightcove Video Connect plugin
3.- Create a new Quiz 
4.- Go to text or any other tab where TinyMCE is used
5.- Editor is not usable (see screenshot)

![Screen Shot 2021-03-27 at 17 12 43](https://user-images.githubusercontent.com/31049169/112737365-b0543980-8f1f-11eb-98ba-64e558ee2047.png)

## Feature request
If you are reporting a bug, delete this section. If you are requesting a feature, please explain what you would like to have in the plugin. Please include examples of other systems/software that may have similar features as well as the exact way you would use this feature.

At first we were thinking this was a problem on our end, however the root cause seems to be any file enqueued with the name `tinymce.js` will generate a conflict with the Quiz Master plugin. For example, if another theme or plugin developer includes a file named like this, the problem will happen as well for them.

We have debugged a little bit on this and think the root cause is in this line https://plugins.trac.wordpress.org/browser/quiz-master-next/trunk/php/admin/quiz-options-page.php#L60

It looks like the way to prevent this from happening is add TinyMCE as a dependency when the enqueued script needs it: https://developer.wordpress.org/reference/functions/wp_enqueue_script/

**Comment by digitalformula** [comment]
QSM team - can you at least respond?  Paying customers are impacted by this.

**Comment by vikasprogrammer** [comment]
Sorry about the delay, we will respond today.

**Comment by etaditya** [comment]
Hi @oscarssanchez 

I could see that you scripts are enqueued everywhere in wp-admin.
Please use conditional loading for scripts as per your plugins pages.
Find a sample code below. I tried it and its working fine.
This won't further mess with any other plugin or theme

```
public static function admin_enqueue_scripts($hook_prefix) {

		global $wp_version;
		global $bc_accounts;
		**if($hook_prefix == 'brightcove_page_brightcove-sources'){**
		// Use minified libraries if SCRIPT_DEBUG is turned off.
		$suffix = ( defined( 'SCRIPT_DEBUG' ) && SCRIPT_DEBUG ) ? '' : '.min';

		$player_api = new BC_Player_Management_API2();
		$players = $player_api->get_all_players();
```

Please put conditions accordingly

Thanx
QSM Team

**Comment by oscarssanchez** [comment]
Hi  @etaditya ,

Thanks for looking into this, upon further digging, we think your suggestion is something we could definitely improve on in the Brightcove plugin. However, it seems the problem would still be something that should be addressed in the Quiz Master plugin. 

At first, I was thinking the function the scripts are being enqueued with is hooked to `admin_enqueue_scripts`, however, 
it seems to be the callback of `setup_admin_menu()` which is hooked to `admin_menu`.

As per WordPress documentation https://developer.wordpress.org/reference/hooks/admin_enqueue_scripts/ scripts in the admin should be enqueued using this hook.I have checked on my end and it looks like this is indeed the issue, the script being loaded at a different time it normally should.

With the following code in the Quiz Master Plugin, the problem resolves: 

```
add_action( 'admin_enqueue_scripts', 'mytest' );

function mytest() {
	wp_enqueue_script( 'wp-tinymce' );
}
```

Please let us know if this makes sense to you and we appreciate the cooperation.

**Comment by etaditya** [comment]
> Hi @etaditya ,
> 
> Thanks for looking into this, upon further digging, we think your suggestion is something we could definitely improve on in the Brightcove plugin. However, it seems the problem would still be something that should be addressed in the Quiz Master plugin.
> 
> At first, I was thinking the function the scripts are being enqueued with is hooked to `admin_enqueue_scripts`, however,
> it seems to be the callback of `setup_admin_menu()` which is hooked to `admin_menu`.
> 
> As per WordPress documentation https://developer.wordpress.org/reference/hooks/admin_enqueue_scripts/ scripts in the admin should be enqueued using this hook.I have checked on my end and it looks like this is indeed the issue, the script being loaded at a different time it normally should.
> 
> With the following code in the Quiz Master Plugin, the problem resolves:
> 
> ```
> add_action( 'admin_enqueue_scripts', 'mytest' );
> 
> function mytest() {
> 	wp_enqueue_script( 'wp-tinymce' );
> }
> ```
> 
> Please let us know if this makes sense to you and we appreciate the cooperation.

Thanks for this update.
We will look into the issue and take proper actions accordingly.
Also if you feel any issue further with Quiz Master Plugin feel free to contact us

Regards
QSM Team

**Comment by scazorla19** [comment]
Hi QSM team,

Do you have any update with this issue?
There are a number of end-users affected by this and would very much appreciate some feedback.

Please and thank you.

**Comment by etaditya** [comment]
Made proper changes 
Sorry for the delay.
Will be released soon.

Thank You

**Comment by scazorla19** [comment]
@etaditya Thank you for the update
Please keep us informed once you have a release.

**Comment by etaditya** [comment]
Hi @scazorla19 @oscarssanchez ,

Just released QSM 7.1.16
with the fix we were discussing about

Thank you

**Comment by digitalformula** [comment]
Confirmed this appears to be fixed.  Thanks.
